### PR TITLE
Gbe 308: Bulk download of audio tracks per show

### PR DESCRIPTION
**Pull request by jonkiparsky**
Looking into the possibilities of testing, I ran into a number of hurdles that I think we should address, but not now. 

Specifically, these center on the use of file objects in tests (ie, providing an audio track to a test object) and testing actions which alter the state of the file system (ie, testing that an action moves an object in the file system)
Since these are required for basically everything we're doing here, I haven't written any new tests for this issue. Creating new tickets to address this post-event.

### CODE REVIEW

**Comment by bethlakshmi** [comment]
Is something missing from the checkin?

When I run the server and load any page, I get:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/staticfiles/handlers.py", line 67, in __call__
    return self.application(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/wsgi.py", line 187, in __call__
    self.load_middleware()
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 47, in load_middleware
    mw_instance = mw_class()
  File "/usr/local/lib/python2.7/dist-packages/django/middleware/locale.py", line 24, in __init__
    for url_pattern in get_resolver(None).url_patterns:
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 365, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 360, in urlconf_module
    self._urlconf_module = import_module(self.urlconf_name)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/importlib.py", line 40, in import_module
    __import__(name)
  File "/vagrant/expo/expo/urls.py", line 14, in <module>
    url(r'^', include('gbe.report_urls', namespace = 'reporting')),
  File "/usr/local/lib/python2.7/dist-packages/django/conf/urls/__init__.py", line 26, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/importlib.py", line 40, in import_module
    __import__(name)
  File "/vagrant/expo/gbe/report_urls.py", line 35, in <module>
    reporting.refresh_tracks, name='refresh_tracks'),
AttributeError: 'module' object has no attribute 'refresh_tracks'
[19/Jan/2016 02:37:08] "GET / HTTP/1.1" 500 59

```

Even the home page causes this.  It looks like refresh_tracks is referenced in reports_urls.py but not in anything else?

**Comment by jonkiparsky** [comment]
Probably my mistake. I changed the name of that function, and probably didn't push the complete. It should be download_tracks_for_show instead, if you want to try making that change. If not, I'll try to push a fix tonight.

**Comment by jonkiparsky** [comment]
@bethlakshmi fixed

**Comment by jonkiparsky** [comment]
To test: check that saving any audioinfo "unsyncs" the downloads folder (moving all prepared downloads to a trash directory). Check that clicking the "download" link creates a zipped tarball under MEDIA_ROOT/uploade/audio/downloads/ and downloads a copy of same to your machine.

File: expo/gbe/management/commands/sync_audio_downloads.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                            title=show_name).get_acts()
+    workdir = make_safe_filename("%s_%s" % (conf_slug, show_name))
+    os.mkdir(workdir)
+
+
+    for act in acts:
+        if act.audio and act.audio.track:
+            fname = os.path.basename(act.audio.track.path)
+            workname = os.path.join(workdir, fname)
+            shutil.copyfile(fname, workname)
```
Comment (bethlakshmi) [inline expo/gbe/management/commands/sync_audio_downloads.py]: Of note, we should check for the filename before attempting this.  See #675.

Not a blocker

**Pull request merged**